text/pvm_invocations.tex

```diff
@@ -732,34 +734,67 @@ \subsection{Refine Functions}\label{sec:refinefunctions}
   \texttt{poke} = 20 \\
   $g = 10$} &
   $\begin{aligned}
-    \using [n, a, b, l] &= \registers_{7 \dots 11} \\
+    \using [n, s, o, l] &= \registers_{7 \dots 11} \\
+    \using \mathbf{s} &= \begin{cases}
+      \none &\when n \not\in \keys{\mathbf{m}} \\
+      \memr_{s\dots+l} &\when \N_{s\dots+l} \subseteq \mathbb{V}_\memory \wedge \N_{o\dots+l} \subseteq \mathbb{V}^*_{\mathbf{m}[n]_\mathbf{u}} \\
+      \error &\otherwise
+    \end{cases}\\
+    (\registers'_7, \mathbf{m}') &\equiv \begin{cases}
+      (\mathtt{OOB}, \mathbf{m}) &\when \mathbf{s} = \error \\
+      (\mathtt{WHO}, \mathbf{m}) &\when \mathbf{s} = \none \\
+      (\mathtt{OK}, \mathbf{m}')\,,\ \where \mathbf{m}' = \mathbf{m} \exc (\mathbf{m}'[n]_\mathbf{u})_{o\dots+l} = \mathbf{s} &\otherwise \\
+    \end{cases} \\
+  \end{aligned}$\\
+  \cmidrule(lr){1-1}\cmidrule(lr){2-2}
+  \makecell*[l]{
+  $\Omega_Z(\gascounter, \registers, \memory, (\mathbf{m}, \mathbf{e}))$ \\
+  \texttt{zero} = 21 \\
+  $g = 10$} &
+  $\begin{aligned}
+    \using [n, p, c] &= \registers_{7 \dots 10} \\
     \using \mathbf{u} &= \begin{cases}
       \mathbf{m}[n]_\mathbf{u} &\when n \in \keys{\mathbf{m}} \\
       \error &\otherwise\\
     \end{cases} \\
```

Comment:
Zero should return an error if the program tries to allocate any pages in the lower 64k of address space/the lower 16 pages of address space.